### PR TITLE
Fix Unit Tests

### PR DESCRIPTION
**Pull request by Ashvin-Ranjan**
- This should fix all of the tests in the `src/tests` directory, which are currently broken (as mentioned in #48 and #44)
- There also may be changes to other classes such as `Universe` and `Meaning`

### Notes
- These tests will still not work until #47 is merged, since it fixes and issue which blocks the use of the `ultk` package entirely

### CODE REVIEW

**Comment by Ashvin-Ranjan** [comment]
I have currently left this as a draft because there are a few issues which I wanted to get some clarification on before I think this is ok to merge.

The first is that it appears that there used to be some old functionality with `Universe` where if only a tuple of `Referent`s were passed in, then the `prior` tuple would be generated automatically to be a uniform distribution. This seemed to be removed when `@dataclass(frozen=True)` was added to `Universe`. I have added this functionality back by creating a constructor that uses `object.__set_attr__` but if this is too big of a change or unwanted this could be reverted.

The second is that I am unable to currently fix the test `test_meaning_subset` in `test_semantics.py` because it tests a functionality of `Meaning` which appears to have been removed where when `Meaning.__init__` was called it would check to see that all `Referent`s passed in were also contained in the `Universe`. I am wondering if this should be re-implemented or if the test should be deleted.

Finally, for `test_language_degree` in `test_language.py`, I have rewritten the `isAnimal` function, but I am unsure if the current functionality of the test actually tests what it is supposed to.

**Review by shanest — CHANGES_REQUESTED** [review]
Thanks, Ash, great stuff!  A couple small things here and there.  And for the third/final question: `isAnimal`/degree_property looks good to me

File: src/ultk/language/semantics.py [hunk 1]
```diff
@@ -74,6 +74,16 @@ class Universe:
...
     prior: tuple[float, ...]
 
+    def __init__(self, referents, prior = None):
+        # use of __setattr__ is to work around the issues with @dataclass(frozen=True)
+        object.__setattr__(self, "referents", referents)
+        # When only referents are passed in, make the priors a unifrom distribution
+        object.__setattr__(
+            self,
+            "prior",
+            prior[:] if prior is not None else (1.0/len(referents) for _ in referents)
```
[the remark this answers is not in the document]
Comment (Ashvin-Ranjan) [reply src/ultk/language/semantics.py]: Thank you for pointing out the issue with the tuple needing `tuple` before it. I have tested `prior or tuple(1 / len(referents) for _ in referents)` and it does work, so I have used that instead.

**Review by shanest — APPROVED** [review]
Awesome, thanks Ash!

**Pull request merged**